napari/layers/points/_points_mouse_bindings.py

```diff
@@ -139,3 +138,83 @@ def _toggle_selected(selected_data, value):
         selected_data.add(value)
 
     return selected_data
+
+
+def _update_drag_vectors_from_event(layer, event):
+    """Update the drag normal and drag up vectors on the Points layer from
+    a mouse event object.
+
+    Note that in 2D mode, the layer._drag_normal and layer._drag_up
+    are set to None
+
+    Parameters
+    ----------
+    layer : "napari.layers.Points"
+        The Points layer to update
+    event
+        The mouse event object
+    """
+    n_display = len(event.dims_displayed)
+    if n_display == 3:
+        ndim_world = len(event.position)
+        # if in 3D, set the drag normal and up directions
+        # get the indices of the displayed dimensions
+        dims_displayed_data = layer._world_to_data_dims_displayed(
+            dims_displayed=event.dims_displayed, ndim_world=ndim_world
+        )
+
+        # get the view direction in data coordinates
+        layer._drag_normal = np.asarray(
+            layer._world_to_data_ray(event.view_direction)
+        )[dims_displayed_data]
+
+        # get the up direction of the camera
+        layer._drag_up = np.asarray(
+            layer._world_to_data_ray(event.up_direction)
+        )[dims_displayed_data]
+
+    else:
+        # if in 2D, set the drag normal and up to None
+        layer._drag_normal = None
+        layer._drag_up = None
+
+
+def _select_points_from_drag(layer, modify_selection: bool, n_display: int):
+    """Select points on a Points layer after a drag event.
+
+    Parameters
+    ----------
+    layer : napari.layers.Points
+        The points layer to select points on.
+    modify_selection : bool
+        Set to true if the selection should modify the current selected data
+        in layer.selected_data.
+    n_display : int
+        The number of dimensions current being displayed
+    """
+    if len(layer._view_data) > 0:
+        # if there is data in view, find the points in the drag box
```

Comment:
you could drop a level of nesting here by doing 

```suggestion
    if len(layer._view_data) > 0:
        layer.selected_data = set()
        return
```

I find the above a little more readable, keeps the code specific to each case a bit more together

Reply:
... should that be a `== 0`, @alisterburt?

Reply:
Good point @alisterburt . I've implemented it as `if len(layer._view_data) == 0` as @jni suggested.